java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java

```diff
@@ -136,6 +136,6 @@ public synchronized void reset() throws IOException {
 
   @Override
   public long skip(long byteCount) throws IOException {
-    throw new UnsupportedOperationException();
+    return super.skip(byteCount);
```

Comment:
I think the same. It needs to do exactly what read() does except maybe for copying the decrypted chunks into the final buffer...